### PR TITLE
Remove special handling of ID for server xlator

### PR DESCRIPTION
**Pull request by prashanthpai**
The xlator ID should default to name of its .so file unless the xlator
defines 'xlator_api_t' structure which has the 'identifier' field. This
was being specially handled for protocol/server xlator. As 'identifier'
field of server xlator has now been been renamed to 'server' from
'server-protocol' in glusterfs source we don't need special handling
anymore:

Reference: https://review.gluster.org/#/c/20490/

Signed-off-by: Prashanth Pai <ppai@redhat.com>

### CODE REVIEW

**Comment by prashanthpai** [comment]
Please don't merge this PR until https://review.gluster.org/#/c/20472/ (merged) gets into a glusterfs nightly RPM which will happen by tomorrow.

File: glusterd2/xlator/load.go [hunk 1]
```diff
@@ -83,17 +83,14 @@ func loadXlator(xlPath string) (*Xlator, error) {
...
 	xlSym := C.CString("xlator_api")
 	defer C.free(unsafe.Pointer(xlSym))
 
 	p := C.dlsym(handle, xlSym)
 	if p != nil {
 		xp := (*C.xlator_api_t)(p)
-		// FIXME: It's named "server-protocol" instead of "server" in server.so
-		//        https://review.gluster.org/18879
-		// xl.ID = C.GoString(xp.identifier)
+		xl.ID = C.GoString(xp.identifier)
```
Comment (Madhu-1) [inline glusterd2/xlator/load.go]: question:
can we add the check here itself, in simplified state

```
if xl.ID = C.GoString(xp.identifier);xl.ID==""{
                xl.ID = strings.TrimSuffix(filepath.Base(xlPath),
		filepath.Ext(xlPath))
}
```

Comment (prashanthpai) [reply glusterd2/xlator/load.go]: The code below is common to both if and else part.

**Comment by prashanthpai** [comment]
retest this please

**Comment by prashanthpai** [comment]
retest this please

**Comment by prashanthpai** [comment]
retest this please

**Comment by prashanthpai** [comment]
retest this please

**Comment by prashanthpai** [comment]
retest this please

**Pull request merged**